x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -201,6 +195,31 @@ protected Expression regexToEquals(RegexMatch<?> regexMatch, Literal literal) {
         }
     }
 
+    private static class EqlBooleanSimplification extends BooleanSimplification {
+
+        EqlBooleanSimplification() {
+            super();
+        }
+
+        @Override
+        protected Expression simplifyNot(Not n) {
```

Comment:
Since the code is similar to the parent, extract the negateable condition in a separate protected method and simply override that in the subclass.